src/ampycloud/data.py

```diff
@@ -514,9 +662,65 @@ def find_slices(self) -> None:
             pass
 
         # Finally, let's metarize these slices !
-        self.metarize(which='slices', base_lvl_alt_perc=self.prms['BASE_LVL_ALT_PERC'],
-                      base_lvl_lookback_perc=self.prms['BASE_LVL_LOOKBACK_PERC'],
-                      lim0=self.prms['MAX_HITS_OKTA0'], lim8=self.prms['MAX_HOLES_OKTA8'])
+        self.metarize(
+            which='slices',
+        )
+
+    def _merge_close_groups(self) -> None:
+        """Merge groups that are closer than the minimum separation at their
+        respective altitudes.
+
+        Algorithm steps:
+            1. Calculate preliminery group base altitude
+            2. Start from bottom and search for the first two groups that are
+            too close.
+            3. Merge these groups and recalculate all base altitudes.
+            4. Repeat 2. and 3. until all groups are enough separated.
+
+        The reason this is done iteratively is that there can be multiple overlaps.
+        In these cases merging two groups often separates their new (combined)
+        base altitude far enough from the base altitude of the next layer above.
+        Doing the merging all at once would thus lead to unwanted overgrouping.
+
+        """
+        # Let's setup a groups df and get the groups base altitude.
+        # setup pd.DataFrame to store slices/ groups/ layers
+        prelim_groups, oids = self._setup_sligrolay_pdf('groups')
+        # calculate bae altitude
+        prelim_groups = self._calculate_sligrolay_base_height(
+            'groups', prelim_groups, oids
+        )
+        # Ordering by altitude
+        prelim_groups.sort_values('alt_base', inplace=True)
+        prelim_groups.reset_index(drop=True, inplace=True)
+        #self.metarize('groups')
+        #prelim_groups = self.groups
+
+        min_seps_grp = prelim_groups['alt_base'].apply(self._get_min_sep_for_altitude)
+        base_alt_diffs = prelim_groups['alt_base'].diff()
+        # create a boolean series to select values, fillna is necessary as
+        # first entry of diff will be nan
+        lt_min_sep_indexer = (base_alt_diffs < min_seps_grp).fillna(False)
+        while len(prelim_groups[lt_min_sep_indexer]) > 0:
+            # start with the two lowest layers that are too close
+            idx = prelim_groups[lt_min_sep_indexer].index[0]
+            # set the group id in the ceilo data to id of group below
+            data_idxer = self.data['group_id'] == prelim_groups['original_id'].loc[idx]
+            self.data.loc[data_idxer, 'group_id'] = prelim_groups['original_id'].iloc[idx - 1]
+            # drop the group
+            prelim_groups.drop(index=idx, inplace=True)
+            # resetting because we must not have index gaps in the next iteration
+            prelim_groups.reset_index(drop=True, inplace=True)
+            # now we recalculate the base alt for the merged supergroup
```

Comment:
Should we recalculate the amount too?

Reply:
I think, it is not necessary here, as we call the full `metarize` method after the merging step.